exa/core/box.py

```diff
@@ -21,8 +23,20 @@ class Box(Base):
     meta = Dict(help="metadata dictionary")
     description = Unicode(help="container description")
 
-    def copy(self, **kws):
-        return self.__class__(**kws)
+    def copy(self, *args, **kws):
+        """Copy the box and all its Data. All args and kwargs
+        are forwarded to the underlying Data.data().copy()
+        method and assumes a deepcopy for each Data.
+
+        Warning:
+            If dataset is large, may impact performance.
+            Are you sure you need a copy?
+        """
+        copy = {}
+        for name, data in self._data.items():
+            # TODO : expose Data.copy args, kwargs?
```

Comment:
This is an interesting idea. I see the value: individual data objects may require specific options on copy. I can't think of any options right now though that can't be handled by smart defaults. And the downside is that the implementation/logic would be more complicated than it currently is. The downside of the current approach is that the `args, kws` are required to be valid arguments for every data object's `.copy` method. I would consider staring out with limited functionality: only allow `args, kws` for the class constructor and use the individual data objects' default `.copy` methods.

Reply:
Yeah I think quite a bit of this is up for change. Honestly I think the usefulness of a `copy` method on such a container object must be met with some valid use cases before we can settle on a good API.